examples/ConstraintIB/eel2d/example.cpp

```diff
@@ -261,26 +308,101 @@ run_example(int argc, char* argv[])
         // Main time step loop.
         double loop_time_end = time_integrator->getEndTime();
         double dt = 0.0;
+        double current_time, new_time;
+        double box_disp = 0.0;
         while (!MathUtilities<double>::equalEps(loop_time, loop_time_end) && time_integrator->stepsRemaining())
         {
             iteration_num = time_integrator->getIntegratorStep();
             loop_time = time_integrator->getIntegratorTime();
+            current_time = loop_time;
 
             pout << "\n";
             pout << "+++++++++++++++++++++++++++++++++++++++++++++++++++\n";
             pout << "At beginning of timestep # " << iteration_num << "\n";
             pout << "Simulation time is " << loop_time << "\n";
 
             dt = time_integrator->getMaximumTimeStepSize();
-            time_integrator->advanceHierarchy(dt);
             loop_time += dt;
+            new_time = loop_time;
+
+            // Regrid the hierarchy if necessary
+            if (time_integrator->atRegridPoint()) time_integrator->regridHierarchy();
+
+            // Set the box velocity to nonzero only if the eel has moved sufficiently far
```

Comment:
This is done to make sure that the body remains inside the control volume. Basically user knows what is size of the control volume at the beginning at should bear the responsibility of moving it to enclose the body at all times. It can be automated, but I think it quite easy to change the translation of control volume in main.cpp as deemed appropriate by the user.